src/components/Badge/Badge.tsx

```diff
@@ -1,61 +1,53 @@
 import * as React from 'react';
 import {oneOf, node} from 'prop-types';
-import {Badge as CoreBadge} from 'wix-ui-core/Badge';
-import {ThemedComponent} from 'wix-ui-theme';
+import {Badge as CoreBadge, BadgeProps as CoreBadgeProps} from 'wix-ui-core/StylableBadge';
+import style from './Badge.st.css';
+import {withStylable} from 'wix-ui-core/withStylable';
+import {UIText} from '../StylableUIText';
+import {SKIN, TYPE, Type, Skin} from './constants';
 
-import {theme} from './theme';
-import {UIText} from '../UIText';
-import {SKIN, FORM, Form, Skin} from './constants';
+interface BadgeProps {
+  type?: Type;
+  skin?: Skin;
+  prefixIcon?: React.ReactNode;
+  suffixIcon?: React.ReactNode;
+}
 
-const iconStyles = {
-  prefix: {paddingRight: '8px'},
-  suffix: {paddingLeft: '8px'}
+const defaultProps = {
+  type: TYPE.solid,
+  skin: SKIN.default
 };
 
-const createBadgeIcon = (type, icon, size) => (
-  icon ?
-    <span style={iconStyles[type]}>
-      {React.cloneElement(icon, {size})}
-    </span> :
-    null
+const StyledBadge = withStylable<CoreBadgeProps, BadgeProps>(
+  CoreBadge,
+  style,
+  ({skin, type}) => ({skin, type}),
+  defaultProps
```

Comment:
Can't we just pass `CoreBadge.defaultProps`?

Reply:
I am defining here additional bo props - type & skin. I expect the CoreBadge.defaultProps to be handled